apis/openstack-nova/src/test/java/org/jclouds/openstack/nova/v2_0/extensions/AttachInterfaceApiExpectTest.java

```diff
@@ -0,0 +1,89 @@
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableSet;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.openstack.nova.v2_0.NovaApi;
+import org.jclouds.openstack.nova.v2_0.domain.FixedIP;
+import org.jclouds.openstack.nova.v2_0.domain.InterfaceAttachment;
+import org.jclouds.openstack.nova.v2_0.domain.PortState;
+import org.jclouds.openstack.nova.v2_0.internal.BaseNovaApiExpectTest;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "unit", testName = "AttachInterfaceApiExpectTest")
+public class AttachInterfaceApiExpectTest extends BaseNovaApiExpectTest {
+
+   public void testAttachInterfacesList() throws Exception {
+      HttpRequest list = HttpRequest.builder().method("GET")
+            .endpoint("https://az-1.region-a.geo-1.compute.hpcloudsvc.com/v2/3456/servers/instance-1/os-interface")
+            .addHeader("Accept", "application/json").addHeader("X-Auth-Token", authToken).build();
+
+      HttpResponse listResponse = HttpResponse.builder().statusCode(200)
+            .payload(payloadFromResource("/attach_interfaces_list.json")).build();
+
+      NovaApi novaApi = requestsSendResponses(keystoneAuthWithUsernameAndPasswordAndTenantName,
+            responseWithKeystoneAccess, extensionsOfNovaRequest, extensionsOfNovaResponse, list, listResponse);
+
+      FluentIterable<InterfaceAttachment> interfaceAttachments = novaApi.getAttachInterfaceApi("az-1.region-a.geo-1")
+            .get().list("instance-1");
+
+      Optional<? extends InterfaceAttachment> interfaceAttachment = interfaceAttachments.first();
+
+      assertTrue(interfaceAttachment.isPresent(), "Couldn't find interface attachment");
+      assertEquals(interfaceAttachment.get(), testInterfaceAttachment());
+   }
+
+   public void testAttachInterfaceGet() throws Exception {
+      HttpRequest list = HttpRequest
+            .builder()
+            .method("GET")
+            .endpoint(
+                  "https://az-1.region-a.geo-1.compute.hpcloudsvc.com/v2/3456/servers/instance-1/os-interface/ce531f90-199f-48c0-816c-13e38010b442")
+            .addHeader("Accept", "application/json").addHeader("X-Auth-Token", authToken).build();
+
+      HttpResponse listResponse = HttpResponse.builder().statusCode(200)
+            .payload(payloadFromResource("/attach_interface_details.json")).build();
+
+      NovaApi novaApi = requestsSendResponses(keystoneAuthWithUsernameAndPasswordAndTenantName,
+            responseWithKeystoneAccess, extensionsOfNovaRequest, extensionsOfNovaResponse, list, listResponse);
+
+      InterfaceAttachment interfaceAttachment = novaApi.getAttachInterfaceApi("az-1.region-a.geo-1").get()
+            .get("instance-1", "ce531f90-199f-48c0-816c-13e38010b442");
+
+      assertEquals(interfaceAttachment, testInterfaceAttachment());
+   }
```

Comment:
Add also tests for the get and list fallbacks, a test for the create method, and the tests for the delete method and its fallback.